keps/sig-network/2681-pod-host-ip/README.md

```diff
@@ -273,7 +273,7 @@ The feature should continue to work just fine.
 
 ###### Are there any tests for feature enablement/disablement?
 
-No, these will be introduced in the Alpha phase.
+Yes, unit tests and e2e tests.
```

Comment:
Already added in Alpha

https://github.com/kubernetes/kubernetes/blob/2472291790e8142deb063024b413b2c6f9f02ff2/pkg/api/pod/util_test.go#L1255

https://github.com/kubernetes/kubernetes/blob/c0691f378424890b278768255b0f554679bea1f4/pkg/apis/core/validation/validation_test.go#L23077

https://github.com/kubernetes/kubernetes/blob/b2613dd381b5e4e8b26e75ad70818a8819e0de3a/test/e2e_node/pod_host_ips.go#L43

Reply:
These are great feature tests, but these aren't enablement/disablement tests per-se.

As described in the KEP template:
```
The e2e framework does not currently support enabling or disabling feature
gates. However, unit tests in each component dealing with managing data, created
with and without the feature, are necessary. At the very least, think about
conversion tests if API types are being modified.

Additionally, for features that are introducing a new API field, unit tests that
are exercising the `switch` of feature gate itself (what happens if I disable a
feature gate after having objects written with the new field) are also critical.
You can take a look at one potential example of such test in:
https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
```

The second paragraph is related here and (please correct me if I'm wrong) you don't have a test like that.

Reply:
Sorry for delay

Which unit test are you talking about?
I'm specifically looking for tests that switch the value of feature gate _in the middle of the test_, because that's really what enablement/disablement is.

Reply:
This `featureEnabled` controls the on/off switching of features.

https://github.com/kubernetes/kubernetes/blob/83f2d89dc987e152f27b31bf630c58ce855d954d/pkg/api/pod/util_test.go#L1189-L1250
This case has 3 parameters: `oldPodStatus`, `podStatus`, `featureEnabled`, each with only 2 values, and a total of 8 possible switches, all of which are covered.

https://github.com/kubernetes/kubernetes/blob/83f2d89dc987e152f27b31bf630c58ce855d954d/pkg/apis/core/validation/validation_test.go#L23075-L23097
The case is the same, that has 2 parameters: `fieldSel`, `featureEnabled`, each with only 2 values, and a total of 4 possible switches, all of which are covered.

Reply:
OK - thanks for the details.
This is pretty non-standard way of doing that. But I'm fine with this.

Please just link the tests (both the unit and the e2e you created which is great) here.